logstash-core/lib/logstash/compiler/lscl.rb

```diff
@@ -103,10 +103,16 @@ def expr_attributes
         end
       }.reduce({}) do |hash, kv|
         k, v = kv
-        if hash[k].nil?
+        existing = hash[k]
+        if existing.nil?
           hash[k] = v
+        elsif existing.kind_of?(::Hash)
+          # For legacy reasons, a config can contain multiple `AST::Attribute`s with the same name (e.g., "match"
+          # in the grok filter); when the value of this attribute produces a `::Hash`, an attempt is made to silently
+          # merge-smash them together (NOTE: this bypasses `AST::Hash`'s ability to detect duplicate keys)
```

Comment:
@yaauie patch LGTM :) but can we improve this comment a little? Reading this in isolation I wouldn't understand what's going on here tbh.
Maybe we can just spell it out explicitly instead of that long explanation and say:

```ruby
# For legacy reasons, a config can contain multiple `AST::Attribute`s with the same name and a hash type value (e.g., "match" in the grok filter)
# which are merged into a single hash value; E.g. `{"match" => {"baz" => "bar"} "match" => { "foo" => "bulb"  }}` is interpreted as `{"match" => {"baz" => "bar", "foo" => "blub"}}`.   
```

IMO that makes it a lot clearer, wdyt?